third_party/move/move-binary-format/src/compatibility.rs

```diff
@@ -130,19 +141,25 @@ impl Compatibility {
         // we can remove/change a friend function if the function is not used by any module in the
         // friend list. But for simplicity, we decided to go to the more restrictive form now and
         // we may revisit this in the future.
-        for (name, old_func) in &old_module.exposed_functions {
-            let new_func = match new_module.exposed_functions.get(name) {
+        for old_func in old_view.functions() {
+            if old_func.visibility() == Visibility::Private {
+                // Function not exposed, continue with next one
+                continue;
+            }
+            let new_func = match new_view.function_definition(old_func.name()) {
                 Some(new_func) => new_func,
                 None => {
-                    if matches!(old_func.visibility, Visibility::Friend) {
-                        friend_linking = false;
+                    // Function has been removed
+                    if matches!(old_func.visibility(), Visibility::Friend) {
+                        friend_linking_errors.push(format!("fun `{}` removed", old_func.name()));
                     } else {
-                        struct_and_pub_function_linking = false;
```

Comment:
How did removing private functions work before? Or `exposed_functions` are only public/friend function declarations, right?

Reply:
Yes, exposed are public/friend/entry functions, and private is not visited.